identity-account/src/account/account.rs

```diff
@@ -310,7 +314,7 @@ impl Account {
     self.storage().state(self.did()).await?.ok_or(Error::IdentityNotFound)
   }
 
-  pub(crate) async fn process_update(&mut self, update: Update) -> Result<()> {
+  pub async fn process_update(&mut self, update: Update) -> Result<()> {
```

Comment:
I'm hoping we can reverse this. I see this as an internal method and the proper way is through `update_identity`.

Reply:
I already tried that, the problem is that with `attach_relationship` we get a `Vec` from JavaScript (Since we deserialize to `OneOrMany`). And the builder pattern only takes one `MethodRelationship`, If we try to create a loop, we will get an error for variables being dropped (it doesn't happen if you chain them, but we can chain anymore because we need the loop). The solution for that is by setting them in temporary variables, but that got really messy. Plus, creating an instance of `Update` insures that all properties are set.

Reply:
This works fine on my side.

```rust
let promise: Promise = future_to_promise(async move {
      let mut account: RefMut<Account> = account.as_ref().borrow_mut();
      let mut updater: IdentityUpdater<'_> = account.update_identity();
      let mut attach_relationship: AttachMethodRelationshipBuilder<'_> = updater.attach_method_relationship();

      attach_relationship = attach_relationship.fragment(fragment);

      for relationship in relationships {
        attach_relationship = attach_relationship.relationship(relationship);
      }

      attach_relationship
        .apply()
        .await
        .wasm_result()
        .map(|_| JsValue::undefined())
    });
```

> Plus, creating an instance of Update insures that all properties are set.

Yes, that is definitely a good point and a desirable thing to have. But changing a type in some `Update::*` variant on the Rust side automatically changes the type in the builder as well, so we'd get a compiler error for that. If we add a property to the Rust builders, then we'd have to change the Wasm bindings anyway to add support for that property, too. These seem to be the two scenarios where it matters, and I think those are caught in the typical review process. I don't see that as enough of a reason to pollute the Rust API with two ways of updating identities.